messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -1,339 +1,122 @@
-                inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("Removing listener $listenerPackageName"))
-                listeners.remove(listener)
-            }
-        }
-    }
-
-    // Gist Message Observer
-
-    internal fun observeMessagesForUser(skipQueueCheck: Boolean = false) {
-        // Clean up any previous observers
-        observeUserMessagesJob?.cancel()
-        inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("Starting messages timer"))
-        if (!skipQueueCheck) {
-            gistQueue.fetchUserMessages()
-        }
-        observeUserMessagesJob = coroutineScope.launch {
-            try {
-                // Poll for user messages
-                val ticker = ticker(pollInterval, context = this.coroutineContext)
-                for (tick in ticker) {
-                    gistQueue.fetchUserMessages()
-                }
-            } catch (e: CancellationException) {
-                // Co-routine was cancelled, cancel internal timer
-                inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("Messages timer cancelled"))
-            } catch (e: Exception) {
-                inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("Failed to observe messages for user: ${e.message}"))
-            }
-        }
-    }
-
-    internal fun dismissPersistentMessage(message: Message) {
-        val gistProperties = GistMessageProperties.getGistProperties(message)
-        if (gistProperties.persistent) {
-            inAppMessagingManager.dispatch(InAppMessagingAction.DismissPersistentMessage(message))
-            inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("Persistent message dismissed, logging view"))
-            gistQueue.logView(message)
-        }
-        handleGistClosed(message)
-    }
-
-    internal fun handleGistLoaded(message: Message) {
-        for (listener in listeners) {
-            listener.onMessageShown(message)
-        }
-    }
-
-    internal fun handleGistClosed(message: Message) {
-        for (listener in listeners) {
-            listener.onMessageDismissed(message)
-        }
-    }
-
-    private fun handleGistCancelled(message: Message) {
-        for (listener in listeners) {
-            listener.onMessageCancelled(message)
-        }
-    }
-
-    internal fun handleGistError(message: Message) {
-        for (listener in listeners) {
-            listener.onError(message)
-        }
-    }
-
-    internal fun handleEmbedMessage(message: Message, elementId: String) {
-        for (listener in listeners) {
-            listener.embedMessage(message, elementId)
-        }
-    }
-
-    internal fun handleGistAction(
-        message: Message,
-        currentRoute: String,
-        action: String,
-        name: String
-    ) {
-        for (listener in listeners) {
-            listener.onAction(message, currentRoute, action, name)
+    override fun setUserId(userId: String) {
+        if (state.userId == userId) {
```

Comment:
Should we add similar check on route change too? 🤔

Reply:
it was on middleware level, but added here as it would save load